### PR TITLE
[Storage] Support DatalakeGen2 SAS token

### PR DESCRIPTION
**Pull request by blueww**
## Description

Design pass review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1195

This cmdlet will generate SAS in client only, won't send request to server, so don't need ShouldProcess. Exception is added.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Storage/Storage.Management/help/New-AzDataLakeGen2SasToken.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+```
+
+## DESCRIPTION
+The **New-AzDataLakeGen2SasToken** cmdlet generates a Shared Access Signature (SAS) token for an Azure DatalakeGen2 item.
+
+## EXAMPLES
+
+### Example 1: Generate a SAS token with full permission
+```
+PS C:\> New-AzDataLakeGen2SasToken -FileSystem "filesystem1" -Path "dir1/dir2" -Permission racwdlmeop
```
Comment (wyunchi-ms) [inline src/Storage/Storage.Management/help/New-AzDataLakeGen2SasToken.md:38]: Please remove the prefix: `PS C:\>` here.

File: src/Storage/Storage.Management/help/New-AzDataLakeGen2SasToken.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+```
+
+This example generates a DatalakeGen2 SAS token with full permission.
+
+### Example 2: Generate a SAS token with specific StartTime, ExpireTime, Protocal, IPAddressOrRange, by pipeline a datalakegen2 item
+```
+PS C:\> Get-AzDataLakeGen2Item -FileSystem test -Path "testdir/dir2" | New-AzDataLakeGen2SasToken -Permission rw -Protocol Https -IPAddressOrRange 10.0.0.0-12.10.0.0 -StartTime (Get-Date) -ExpiryTime (Get-Date).AddDays(6)
```
Comment (wyunchi-ms) [inline src/Storage/Storage.Management/help/New-AzDataLakeGen2SasToken.md:45]: Please remove the prefix: `PS C:\>` here.

File: src/Storage/Storage.Management/help/New-AzDataLakeGen2SasToken.md.bak [hunk 3]
```diff
@@ -0,0 +1,238 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Storage.dll-Help.xml
```
Comment (wyunchi-ms) [inline src/Storage/Storage.Management/help/New-AzDataLakeGen2SasToken.md.bak:2]: Please remove the `.bak` file here.

File: src/Storage/Storage.Management/help/New-AzDataLakeGen2SasToken.md (hunk 1 above)
Comment (blueww) [reply src/Storage/Storage.Management/help/New-AzDataLakeGen2SasToken.md]: fixed

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**